### PR TITLE
Publish images as received by SPOT

### PR DESCRIPTION
**Pull request by berndgassmann**
Current status of the PR:
- By default, the driver publishes uncompressed images under `/images` (same as before) but also compressed images under `/compressed`.
- if the driver is launched with `uncompress_images:=False`, only compressed images get published. 

## Testing Done

Extended the interface tests where required to ensure the compressed image function is called correctly.
And did some test recordings using the 5 body cameras.

### CODE REVIEW

File: spot_driver/src/api/default_image_client.cpp
Comment (berndgassmann) [reply spot_driver/src/api/default_image_client.cpp:155]: So the published data should depend on how the robot is configured via the provided control interface in the SPOT-provided API. ROS driver should allow the same configuration and the 1:1 forwarding of the data to ROS. 
PS: At least that's what I now did for the native ROS2 interface in the AD-Driving Simulator CARLA: try to provide the rendering output from Unreal Engine 1:1 towards ROS if possible and adapt only where necessary (e.g. adapt data format to pointCloud ros-message).

**Comment by berndgassmann** [comment]
First step: Moved greyscale part into separate PR.
Second step: instead of adding the option to allow the user to select whether to use decompression or not; I will provide an update of this that performs no decompression at all in the driver and leaves that operation to the user outside.
Nevertheless the template code will remain in because the image requested towards SPOT can already be set to compressed or raw. Depending on what SPOT sends over the network the data should be published accordingly. The templates are only there to prevent from manual code duplication. But as long as the image request to SPOT supports both methods this driver should also allow the user to go both paths.

**Comment by berndgassmann** [comment]
Now you can select what you like more:
Either this PR where no processing on the received image data is performed (this is what I would strongly prefer, because that possibility was the reason for this PR), 
or the #358 where the greyscale JPEG is decompressed correctly

**Comment by tcappellari-bdai** [comment]
@berndgassmann thank you for the PRs! We are looking into them now :)

File: spot_driver/src/api/default_image_client.cpp
Comment (khughes-bdai) [reply spot_driver/src/api/default_image_client.cpp]: Hi @berndgassmann,
Our plan with this PR is to add an optional node that will decompress the images in case the user wants that built-in. This will also minimize the impact on our internal projects with Spot that are assuming to receive uncompressed images. I agree that it's best to have the images always published as compressed by default!
As far as I'm aware, there's no way to get the motor currents from Spot. The closest thing to what you mentioned seems to be the topic `/<robot name>/status/feet`, which gives the position of the feet with respect to the body and whether or not they are in contact with the ground. You could reach out to Boston Dynamics support as well and they might have more insight on this.

**Comment by berndgassmann** [comment]
I updated the PR description to cover all what is left in this PR.
I also merged the main branch in and resolved the conflict with the previous commits merged into main where we extended the decompression handling (which is removed by this one again, and introduced by #364.) Or is #364 completely integrating this and this can be cancelled? Didn't understand the flow yet.
But github is not actually able to handle multiple reviews on top of each. And you are using graphite to get the stacking ability back? Good that such tools close the gap (with gitlab that works out of the box, but one had to perform rebasing if I remember correclty; but rebasing felt cleaner that merging main branch into your own one, as the review history just was not blown up that much).

File: spot_driver/src/api/default_image_client.cpp
Comment (berndgassmann) [reply spot_driver/src/api/default_image_client.cpp]: Ah thanks, yes; that's what I am searching for indeed. That should work out. It even has already some estimates on slipping and friction, so I don't need my fuzzy estimators for such. Maybe I'll then try my odometry plus the Kalman filter on top for orientation correction ImuState (and potentially Gps or other fixed absolute position information to correct the position error if I get the muse ;-)

**Comment by khughes-bdai** [comment]
@berndgassmann We opened #364 as we want to continue publishing uncompressed images by default in order not to break any current uses of the repo, but want the option to publish only uncompressed images (ideally specified by a flag in the launchfile). 
I think the general idea would be something like
`ros2 launch spot_driver driver.launch.py ....` -> this would now publish both compressed and uncompressed images as the default option
`ros2 launch spot_driver driver.launch.py uncompress_images:=false ...` -> this would only publish uncompressed images. 
If this is an easy addition for you to make to this PR, I'd be happy to close the other one and merge this one in! Thanks again for all the contributions :)

**Comment by berndgassmann** [comment]
will do so immediately before my "beer o'clock" (is one really saying that in the U.S.?)

**Comment by berndgassmann** [comment]
It compiles and links and the unit test run. I would guess it's also functional, but I haven't time to startup the robot today anymore. You can try it on your side.
Ah and the launch file itsself I didn't touch yet, but the parameter in the code is there and should work.
Maybe you can add that and push the chnages before merging

File: spot_driver/src/api/default_image_client.cpp
Comment (khughes-bdai) [inline spot_driver/src/api/default_image_client.cpp:135]: Having an issue while testing on an RGB spot where I constantly get this error and no images (camera info, depth, compressed, or rgb) get published. Looking into how to resolve this

**Comment by khughes-bdai** [comment]
Added the launch argument, but this doesn't seem to work currently on our RGB Spots. I think it has to do with the format that the RGB images are coming in as. Let me know if it still works on your greyscale robot when you get a chance.

**Comment by berndgassmann** [comment]
With the help of Philip from  FZI here in Karlsruhe having a SPOT with an Arm we were able to solve the issue. The error was by the way from the depth cameras  which I had deactivated in my setup...

**Comment by khughes-bdai** [comment]
Awesome, I'll test this out again on our RGB robots!

**Review by khughes-bdai — APPROVED** [review]
This looks good to me! I just tested it out on our RGB robots and everything was working okay. By default, the driver will now publish both compressed images under `/compressed` and uncompressed under `/images`. If you want only compressed images, you can launch the driver with `uncompress_images:=False`. 
I would like to get one more approval before merging this in as it's pretty big PR, but the functionality is all there :) Thanks again for all the contributions!

**Review by tcappellari-bdai — APPROVED** [review]
LGTM - thanks!

**Pull request merged**
